File: tmp/see_agent/server/routes/health.py
"""Health-check route for the see-agent API."""

from __future__ import annotations

import logging

from fastapi import APIRouter

from see_agent.server.schemas import HealthResponse

logger = logging.getLogger(__name__)

router = APIRouter()


@router.get("/api/health")
async def health_check() -> HealthResponse:
    """Return a simple health status with the API version.

    Returns:
        A JSON object with ``status`` and ``version`` keys.
    """
    return HealthResponse(status="ok", version="0.1.0")

File: see-app/src/server/routes/health.rs
use axum::routing::get;
use axum::{Json, Router};
use serde::Serialize;

use see::consts::VERSION;

#[derive(Serialize)]
struct HealthResponse {
    status: String,
    version: String,
}

async fn health_check() -> Json<HealthResponse> {
    Json(HealthResponse {
        status: "ok".into(),
        version: VERSION.into(),
    })
}

pub fn router() -> Router {
    Router::new().route("/health", get(health_check))
}
